### PR TITLE
handle http basic auth in piped instance

### PR DESCRIPTION
**Pull request by aspacca**
If api url contains http basic auth credentials propagate them to url returned by api

### CODE REVIEW

File: Model/Applications/PipedAPI.swift [hunk 1]
```diff
@@ -229,16 +249,19 @@ final class PipedAPI: Service, ObservableObject, VideosAPI {
         )
     }
 
-    private static func extractID(from content: JSON) -> Video.ID {
-        content.dictionaryValue["url"]?.stringValue.components(separatedBy: "?v=").last ??
-            extractThumbnailURL(from: content)!.relativeString.components(separatedBy: "/")[4]
+    private func extractID(from content: JSON) -> Video.ID {
```
Comment (aspacca) [inline Model/Applications/PipedAPI.swift:252]: this change is need in case proxy is not at the root of the domain

Comment (FireMasterK) [reply Model/Applications/PipedAPI.swift:252]: This breaks HLS livestreams already I think

Comment (aspacca) [reply Model/Applications/PipedAPI.swift:252]: @FireMasterK can you explain?

Comment (aspacca) [reply Model/Applications/PipedAPI.swift:252]: could not reproduce with `/watch?v=4Zl6IB0jjHk`

**Comment by arekf** [comment]
I don't have a local Piped instance to test this. Is this all we need to make it work with http basic auth? I'll try to implement and test with Invidious.

**Comment by aspacca** [comment]
@arekf I've tested different cases and it seems enough.

my setup is
- domain.tld: frontend
- domain.tld/api: api
- domain.tld/proxy: proxy

everything behind http basic auth

in `config.properties` for piped backend there's no user credentials, since that won't work in the browser. it relies instead on the browser caching credentials for the same domain in the same session

only one thing: @FireMasterK was mentioning the login in Piped is implemented through http basic auth, since the app doesn't support login on Piped I haven't tested this

**Comment by arekf** [comment]
@aspacca I'm adding support for Piped accounts here: https://github.com/yattee/app/pull/2

I cherry-picked your commit on top of the other PR. Could you kindly test and see if it works fine with your case?

**Comment by aspacca** [comment]
I see you added `Accounts` feature to Piped as well.
The problem is that authentication in Piped uses `auhtorization` header, so it won't work with http basic auth.
I discussi with @FireMasterK how to overcome this

also it seems I think you removed some of the credentials propagation, because I cannot play videos anymore.
I got the following alert: "You do not have permission to access the requested resource."

**Comment by tristan-k** [comment]
I would appreciate if support for basic http auth would be added. Right now it is not possible to run an individous instance private because it is not possible to restrict access to registered users only. See [this issue](https://github.com/iv-org/invidious/issues/446) and [this PR](https://github.com/iv-org/invidious/pull/3728) for reference.

**Comment by stonerl** [comment]
I closed this on Since it almost 3 years old. Also we now support this vor invidious. As soon as Piped is working again I'll implement it for it as well.

**Comment by awefdev** [comment]
I just stumbled on this PR; Piped seems to be working again (not sure when that started). Any chance this is being worked on/planned or would a PR be accepted?

**Pull request closed without merging**